### PR TITLE
Test updates for dropping the constant keyword.

### PR DESCRIPTION
**Pull request by ekpyron**
Part of #4090.

Removes further occurances of the constant keyword in the test suite.

### CODE REVIEW

File: test/compilationTests/corion/schelling.sol [hunk 1]
```diff
@@ -417,7 +417,7 @@ contract schelling is module, announcementTypes, schellingVars {
             
         setVoter(msg.sender, voter);
     }
-    function checkReward() public constant returns (uint256 reward) {
+    function checkReward() public returns (uint256 reward) {
```
Comment (chriseth) [inline test/compilationTests/corion/schelling.sol]: Why was it not possible to make this `view`?

File: test/compilationTests/corion/schelling.sol [hunk 2]
```diff
@@ -496,15 +496,15 @@ contract schelling is module, announcementTypes, schellingVars {
         
         require( moduleHandler(moduleHandlerAddress).transfer(address(this), msg.sender, funds, true) );
     }
-    function getCurrentSchellingRoundID() public constant returns (uint256) {
+    function getCurrentSchellingRoundID() public returns (uint256) {
```
Comment (chriseth) [inline test/compilationTests/corion/schelling.sol]: Why not `view`?

File: test/compilationTests/corion/schelling.sol [hunk 3]
```diff
@@ -496,15 +496,15 @@ contract schelling is module, announcementTypes, schellingVars {
...
         /*
             Number of actual Schelling round.
             
             @uint256        Schelling round.
         */
         return getCurrentSchellingRound();
     }
-    function getSchellingRound(uint256 id) public constant returns (uint256 expansion) {
```
Comment (chriseth) [inline test/compilationTests/corion/schelling.sol:507]: Why not `view`?

File: test/compilationTests/zeppelin/crowdsale/Crowdsale.sol [hunk 4]
```diff
@@ -92,15 +92,15 @@ contract Crowdsale {
   }
 
   // @return true if the transaction can buy tokens
-  function validPurchase() internal constant returns (bool) {
+  function validPurchase() internal returns (bool) {
```
Comment (chriseth) [inline test/compilationTests/zeppelin/crowdsale/Crowdsale.sol]: Why not view?

**Review by chriseth — COMMENTED** [review]
Please only add new commits if you make modifications.

**Comment by ekpyron** [comment]
@chriseth I think the respective functions only weren't ``view`` because they called other functions that weren't ``view``, but it should be possible to restrict all of them to ``view`` (see latest commit).

**Review by chriseth — DISMISSED** [review]

**Comment by ekpyron** [comment]
Squashed the last two commits after approving review.

File: test/compilationTests/corion/provider.sol [hunk 5]
```diff
@@ -469,7 +469,7 @@ contract provider is module, safeMath, announcementTypes {
         delete clients[msg.sender].providerConnected;
         emit EClientLost(msg.sender, provider, currHeight, bal);
     }
-    function checkReward(address addr) public constant returns (uint256 reward) {
+    function checkReward(address addr) public returns (uint256 reward) {
```
Comment (erak) [inline test/compilationTests/corion/provider.sol:472]: Can this also be ``view``?

Comment (ekpyron) [reply test/compilationTests/corion/provider.sol:472]: That one depends on ``getProviderReward``, which writes to ``providers`` and therefore can't be ``view``.

**Comment by chriseth** [comment]
Sorry, needs to be rebased.

**Review by chriseth — APPROVED** [review]

**Pull request merged**